lib/Runtime/Library/JsBuiltIn/JsBuiltIn.js

```diff
@@ -238,5 +241,157 @@
 
         return a;
     });
-  
+
+    platform.registerChakraLibraryFunction("FlattenIntoArray", function(target, source, sourceLen, start, depth)
+    {
+        "use strict";
+        //1. Let targetIndex be start.
+        let targetIndex = start;
+        //2. Let sourceIndex be 0.
+        let sourceIndex = 0;
+        //3. Repeat, while sourceIndex < sourceLen
+        let element;
+        while (sourceIndex < sourceLen) {
+            // a. Let P be ! ToString(sourceIndex).
+            // b. Let exists be ? HasProperty(source, P).
+            if (sourceIndex in source) {
+                // c. If exists is true, then
+                //  i. Let element be ? Get(source, P).
+                //  ii. If mapperFunction is present - skipped see separate function
+                //  iii. Let shouldFlatten be false.
+                //  iv. If depth > 0, then
+                //      1. Set shouldFlatten to ? IsArray(element).
+                if (depth > 0 && __chakraLibrary.isArray(source[sourceIndex])) {
+                    // v. If shouldFlatten is true, then
+                    //  1. Let elementLen be ? ToLength(? Get(element, "length")).
+                    //  2. Set targetIndex to ? FlattenIntoArray(target, element, elementLen, targetIndex, depth - 1).
+                    element = source[sourceIndex];
+                    targetIndex = __chakraLibrary.FlattenIntoArray(target, element, element.length |0, targetIndex, depth - 1);
```

Comment:
I believe this operation (`element.length |0`) is not doing the same thing as ToLength. Seems the operation is closer to ToInteger since it does not clamp negative values to 0 like ToLength does.